### PR TITLE
doc: document INSIDE_DUNE in hacking.rst

### LINKED ISSUE #6280: Document `INSIDE_DUNE` env var
I've noticed that `INSIDE_DUNE` isn't documented anywhere, I think it could be nice to have it somewhere in the manual.

### PR DESCRIPTION
**Pull request by Alizter**
- fix https://github.com/ocaml/dune/issues/6280

### CODE REVIEW

File: doc/hacking.rst [hunk 1]
```diff
@@ -98,6 +98,19 @@ For integration tests, we use a system similar to `Cram tests
 
 .. _ppx_expect:      https://github.com/janestreet/ppx_expect
 
+When running dune inside tests, the ``INSIDE_DUNE`` environment variable is set.
+This has the following effects:
+
+* Change the default root detection behaviour to use the current directory
+  rather than the top most ``dune-project`` / ``dune-workspace`` file.
+* Be less verbose various times Dune outputs a user message.
```
Comment (christinerose) [inline doc/hacking.rst:106]:
```suggestion
* Be less verbose in the various times Dune outputs a user message.
```
…or “during the various times” ??
Feels like there’s something missing between “verbose” and “various”

File: doc/hacking.rst [hunk 2]
```diff
@@ -98,6 +98,19 @@ For integration tests, we use a system similar to `Cram tests
...
+* Error reporting is deterministic by default.
+* Prefer not to use a diff program for displaying diffs.
+
+This list is not exhaustive, and may change in the future. In order to find the
```
Comment (christinerose) [inline doc/hacking.rst:110]:
```suggestion
This list is not exhaustive and may change in the future. In order to find the
```

**Review by christinerose — COMMENTED** [review]
A question about some possible missing words and a minor grammar fix.

**Review by emillon — APPROVED** [review]

File: doc/hacking.rst (hunk 1 above)
Comment (Alizter) [reply doc/hacking.rst:106]: Hmm, sounds ok when I say it out loud, but I am no expert on grammar.

Comment (Alizter) [reply doc/hacking.rst:106]:
```suggestion
* Be less verbose when Dune outputs a user message.
```
tada this is better.

Comment (christinerose) [reply doc/hacking.rst]: Yes! Perfect!

**Comment by Alizter** [comment]
@emillon this can be merged now

**Pull request merged**